siaskynet/skynet.py

```diff
@@ -36,13 +40,17 @@ def UploadFile(path, opts=None):
 
         with open(path, 'rb') as f:
             r = requests.post("%s/%s/%s" % (opts.portalUrl, opts.portalUploadPath, uuid), files={opts.portalFileFieldname: f})
-            response = r.json()
-            return "sia://" + response["skylink"]
+            return r
 
     @staticmethod
     def DownloadFile(path, skylink, opts=None):
+        r = Skynet.GetDownloadBuffer(skylink, opts)
```

Comment:
Any reason you've called it `GetDownloadBuffer`?
If not, I think having the interface be
```
+ UploadFile(path, opts=None)
+ PostUploadFile(path, opts=None)
+ DownloadFile(path, skylink, opts=None):
+ GetDownloadFile(skylink, opts=None):
```
would be more consistent. Or perhaps `UploadFileRequest` and `DownloadFileRequest`...

Reply:
cc @Kcchouette

Reply:
No reason, just to differentiate each methods.
At first I would rename the first two methods:
UploadFile →UploadAndSaveFile
DownloadFile →DownloadAndSaveFile
But it would broke you (simple) API on your documentation.

I will change in a few hours to like you want :)
UploadFileRequest / DownloadFileRequest seems good for me

Reply:
↑ cc @peterjanbrone

Reply:
done @peterjanbrone